app/javascript/controllers/selection_controller.js

```diff
@@ -0,0 +1,47 @@
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  static targets = ["rowSelection"];
+  static values = {
+    storageKey: {
+      type: String,
+      default: location.protocol + "//" + location.host + location.pathname,
+    },
+  };
+
+  connect() {
+    this.element.setAttribute("data-controller-connected", "true");
+
+    if (sessionStorage.getItem(this.storageKeyValue)) {
+      this.rowSelectionTargets.map((row) => {
+        if (
+          new Map(JSON.parse(sessionStorage.getItem(this.storageKeyValue))).get(
+            row.value
```

Comment:
I think the bigger reasoning to use an `Array` instead of the `Map` is that we are only using the map to store a boolean state (i.e. checked or not checked).

My original idea of using a Map was not to maintain unchecked items in it. I.e. if the checkbox was unchecked we delete the key from the map and if it is checked we add the key to the map if it doesn't exist in it. Using the Map to store the checked state as a boolean instead makes it a little hard to use local storage later on. I.e. you have to check the value to decide to include it or not.

Thinking a bit more about this, an `Array` is probably the nicer solution. Yes it makes the js code a little less simple in that we have to decide to `splice` the array to remove items, after using `indexOf` to determine the location of the item. But then in other methods that need to use the selection we simply have an `Array` that can likely be sent to the server without any modification. Hopefully that all makes sense.

Reply:
In this case presence in the `Array` would mean that the row is checked and absence would mean that the row is unchecked.

Reply:
Please see 02bb7249f8df9fc1fbbb4a15977bcbbcb43a4c45.